packages/generator/src/options-parser.ts

```diff
@@ -0,0 +1,174 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { Options } from 'yargs';
+import { ParsedOptions } from './generator-options';
+const logger = createLogger('generator-options');
+
+type Option = Options & {
+  replacedBy?: string;
+  describe: string;
+};
+
+/**
+ * @internal
+ * Remove defaults from CLI options. This is necessary to handle default setting on our own.
+ * @param options - CLI options, record mapping option name to option config
+ * @returns CLI options without default values
+ */
+export function getOptionsWithoutDefaults<
+  CliOptionsT extends Record<string, Option>
+>(options: CliOptionsT): CliOptionsT {
+  return Object.entries(options).reduce((opts, [name, option]) => {
+    const { default: def, ...optionWithoutDefault } = option;
+    const describe =
+      def === undefined
+        ? optionWithoutDefault.describe
+        : `${optionWithoutDefault.describe}\n[Default: ${def}]`;
+    return {
+      ...opts,
+      [name]: {
+        ...optionWithoutDefault,
+        describe
+      }
+    };
+  }, {} as CliOptionsT);
+}
+
+/**
+ * @internal
+ * Parse options for programmatic and CLI use.
+ * Warn, if deprecated options are used or duplicate.
+ * Adds defaults if unset.
+ * @param options - Available CLI options along with their configuration.
+ * @param userOptions - Options as set by user.
+ * @returns Parsed options with default values.
+ */
+export function parseOptionsWithDefaults<
```

Comment:
[pp] I would remove the `withDefaults` from the method name and include it in the return type. The method you called is gone after the invocation. However, the types remains and would be `ParseOptionsWithDefault` you could perhaps even argue the this is clear since all properties are not optional on the parse type.

Reply:
I think `ParsedOptions` already implies that the defaults are in there, because this is the result of the parsing. I can remove this from the name and mention it only in the comment. I think that is legit.